src/components/LanguageSwitcher/LanguageSwitcher.tsx

```diff
@@ -2,39 +2,39 @@ import React, { useState } from 'react';
 import Button from '../UI/Button';
 import localIcon from '../../assets/icons/local-icon.svg';
 import { useLocalization } from '../../context/localization-context';
-import { Language } from '../../models/i18n';
+import { Language } from '../../models/localizationt';
 
 const LanguageSwitcher = () => {
-  const { i18n, language, changeLanguage } = useLocalization();
+  const { dataLang, language, changeLanguage } = useLocalization();
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
 
   const toggleDropdown = () => {
     setIsDropdownOpen(!isDropdownOpen);
   };
 
-  const handleLanguage = (languageCode: Language) => {
-    changeLanguage(languageCode);
-    toggleDropdown();
+  const handleLanguageClick = (event: React.MouseEvent<HTMLElement>) => {
+    const target = event.target as HTMLElement;
```

Comment:
The previous solution was correct. You don't have to write handlers this way, React does the same thing behind the scene.